src/state_machine.zig

```diff
@@ -2471,6 +2554,21 @@ pub fn StateMachineType(
     };
 }
 
+// TODO(client_release_min): When client_release_min is bumped, remove this function and the
+// legacy code it gates.
```

Comment:
Good idea! Though it would actually be something like:

```
fn forbid_zero_amounts(client_release: vsr.Release) bool {
    const release_min_inclusive = vsr.Release.from(.{ .major = 0, .minor = 15, .patch = 3 });
    const release_max_exclusive = vsr.Release.from(.{ .major = 0, .minor = 16, .patch = 0 });
    const release_max_transition = vsr.Release.from(.{ .major = 0, .minor = 17, .patch = 0 });

    assert(client_release.value < release_max_transition.value);
    return client_release.value >= release_min_inclusive.value and
        client_release.value < release_max_exclusive.value;
}
```

Reply:
Can we comptime this assert somehow? :thinking: